### PR TITLE
Merge nvidia-gpu-device-plugin and nvidia-device-plugin.

### LINKED ISSUE #19114: Advice user to use --gpus instead of gpu-addon on docker driver
### What Happened?

I am trying to integrate GPU with Minikube, My host machine is Windows and I am using docker desktop in wsl2 and trying to provision Minikube with the support of GPU.

I was able to integrate GPU with docker, 

` sudo docker run --rm --runtime=nvidia --gpus all ubuntu nvidia-smi`

image

- Minikube start command 

`minikube start --driver=docker --addons=ingress --addons=nvidia-gpu-device-plugin`
image

image

- Pod failing with the following error 
```

  Warning  FailedScheduling  45s   default-scheduler  0/1 nodes are available: 1 Insufficient nvidia.com/gpu. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod.
 ```

### Attach the log file

[log.txt](https://github.com/user-attachments/files/15927874/log.txt)

### Operating System

Windows

### Driver

Docker

### step I followed 

[Medium Link](https://medium.com/@sivanaikk0903/kubernetes-with-gpu-for-mlops-workloads-c684f8c8d41c)

### PR DESCRIPTION
**Pull request by RamBITS-AI**
Fixes #19114.

**EDIT:**
**_Merged both the addons, nvidia-gpu-device-plugin and nvidia-device-plugin, into nvidia-device-plugin addon._**

**BEFORE**:

minikube start --driver docker --container-runtime docker --gpus all

PRE-1-minikube-nvidia-gpu-all_docker-driver_container-runtime-docker

minikube status
minikube kubectl -- get pods -A

PRE-1-minikube-nvidia-gpu-all_docker-driver_container-runtime-docker_minikube-status_get-pods

minikube addons enable nvidia-device-plugin --alsologtostderr

PRE-1-minikube-nvidia-device-plugin_addons-enable

minikube addons enable nvidia-gpu-device-plugin --alsologtostderr

PRE-1 1-minikube-nvidia-gpu-device-plugin_addons-enable
PRE-1 2-minikube-nvidia-gpu-device-plugin_addons-enable

**AFTER**:

minikube start --driver docker --container-runtime docker --gpus all

POST-2 1

minikube status
minikube kubectl -- get pods -A

POST-2 2

minikube addons enable nvidia-device-plugin --alsologtostderr

POST-2 3

minikube addons enable nvidia-gpu-device-plugin --alsologtostderr

POST-2 4

### CODE REVIEW

**Review by spowelljr — CHANGES_REQUESTED** [review]
Hi @RamBITS-AI, thanks for the PR! There might have been a miscommunication but `nvidia-device-plugin` is the addon we want users to be using as it's being kept up to date, `nvidia-gpu-device-plugin` on the other hand hasn't been updated in 2.5 years, to that's the one we want to deprecate and move users off of.

**Comment by RamBITS-AI** [comment]
Hi @spowelljr,

Have updated the AFTER screenshots with the latest ones.

File: deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl [hunk 1]
```diff
@@ -29,6 +29,10 @@ spec:
         name: nvidia-device-plugin-ds
     spec:
       tolerations:
+      - operator: Exists
+        effect: NoExecute
+      - operator: Exists
+        effect: NoSchedule
```
Comment (spowelljr) [inline deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl:35]: Curious where you got these YAML changes from

Comment (RamBITS-AI) [reply deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl:35]: From /deploy/addons/gpu/nvidia-gpu-device-plugin.yaml.tmpl - https://github.com/kubernetes/minikube/blob/a0f915eda8a31049eb26e6eaa3dd0133e507bce0/deploy/addons/gpu/nvidia-gpu-device-plugin.yaml.tmpl#L35

Comment (spowelljr) [reply deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl:35]: That config file is very old, those values are likely no longer needed when using the updated image. I think we should leave the config as is (remove the added values).

Comment (RamBITS-AI) [reply deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl]: Ok. Done.

**Comment by RamBITS-AI** [comment]
Hi @spowelljr 

The corrections have been made.

File: deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl [hunk 2]
```diff
@@ -43,14 +43,23 @@ spec:
...
           mountPath: /var/lib/kubelet/device-plugins
+        - name: dev
+          mountPath: /dev
       volumes:
       - name: device-plugin
         hostPath:
           path: /var/lib/kubelet/device-plugins
+      - name: dev
+        hostPath:
+          path: /dev
```
Comment (spowelljr) [inline deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl:65]: Can these be removed as well please?

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl (hunk 2 above)
Comment (RamBITS-AI) [reply deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl:65]: Sure. @spowelljr.

Comment (RamBITS-AI) [reply deploy/addons/nvidia-device-plugin/nvidia-device-plugin.yaml.tmpl]: This is done now.

**Review by spowelljr — APPROVED** [review]
Thanks for the PR!

**Pull request merged**